fleetspeak/src/inttesting/integrationtest/clone.go

```diff
@@ -0,0 +1,265 @@
+	}
+
+	cl, err := client.New(
+		conf,
+		client.Components{
+			ServiceFactories: map[string]cservice.Factory{"NOOP": cservice.NOOPFactory},
+			Communicator:     &chttps.Communicator{},
+		})
+	if err != nil {
+		t.Fatalf("Unable to start intial client: %v", err)
+	}
+	cl.Stop()
+
+	wb, err := ioutil.ReadFile(path.Join(configPath, "writeback"))
+	if err != nil {
+		t.Fatalf("Unable to read example writeback file: %v", err)
+	}
+	for _, pc := range []string{"client_2", "client_3", "client_4", "client_5"} {
+		configPath := path.Join(tmpDir, pc)
+		if err := os.Mkdir(configPath, 0777); err != nil {
+			t.Fatalf("Unable to create client directory [%v]: %v", configPath, err)
+		}
+		ioutil.WriteFile(path.Join(configPath, "writeback"), wb, 0644)
+	}
+
+	// We have 5 config dirs with the same key. Start a client running against
+	// each of them.
+
+	for i, pc := range []string{"client_1", "client_2", "client_3", "client_4", "client_5"} {
+		conf := config.Configuration{
+			ConfigurationPath: path.Join(tmpDir, pc),
+			TrustedCerts:      x509.NewCertPool(),
+			ClientLabels: []*fspb.Label{
+				{ServiceName: "client", Label: "clone_test"},
+			},
+			FixedServices: []*fspb.ClientServiceConfig{{Name: "NOOP", Factory: "NOOP"}},
+			Servers:       []string{listener.Addr().String()},
+			CommunicatorConfig: &clpb.CommunicatorConfig{
+				MaxPollDelaySeconds:    2,
+				MaxBufferDelaySeconds:  1,
+				MinFailureDelaySeconds: 1,
+			},
+		}
+		if !conf.TrustedCerts.AppendCertsFromPEM(cert) {
+			t.Fatal("unable to add server cert to pool")
+		}
+		cl, err := client.New(
+			conf,
+			client.Components{
+				ServiceFactories: map[string]cservice.Factory{"NOOP": cservice.NOOPFactory},
+				Communicator:     &chttps.Communicator{},
+			})
+		if err != nil {
+			t.Fatalf("Unable to start client %d: %v", i+1, err)
+		}
+		defer cl.Stop()
+	}
+
+	var id common.ClientID
+
+	// Wait for one client id to show up through the admin interface.
+	ctx, fin := context.WithTimeout(context.Background(), 20*time.Second)
+	for {
+		res, err := admin.ListClients(ctx, &spb.ListClientsRequest{})
+		if err != nil {
+			t.Fatalf("Unable to list clients: %v", err)
+		}
+		if len(res.Clients) > 1 {
+			t.Fatalf("Only expected 1 client, got %d", len(res.Clients))
+		}
+		if len(res.Clients) == 1 {
+			cid, err := common.BytesToClientID(res.Clients[0].ClientId)
+			if err != nil {
+				t.Fatalf("Unable to parse listed client id: %v", err)
+			}
+			id = cid
+			break
+		}
+		select {
+		case <-ctx.Done():
+			t.Fatalf("Timed out waiting for client id: %v", ctx.Err())
+		case <-time.After(200 * time.Millisecond):
+			continue
+		}
+	}
+	fin()
+
+	// Wait for 20 contacts.
+	ctx, fin = context.WithTimeout(context.Background(), 20*time.Second)
+	for {
+		res, err := admin.ListClientContacts(ctx, &spb.ListClientContactsRequest{ClientId: id.Bytes()})
+		if err != nil {
+			t.Fatalf("Unable to list client contacts: %v", err)
+		}
+		if len(res.Contacts) > 20 {
```

Comment:
nit

Strictly speaking, this should be >= 20: We are waiting for at least 20 contacts..

Reply:
Done.